.github/workflows/main.yml

```diff
@@ -1,30 +1,112 @@
-# This is a basic workflow to help you get started with Actions
-
 name: CI
 
-# Controls when the action will run. 
 on:
-  # Triggers the workflow on push or pull request events but only for the master branch
   push:
-    branches: [ master ]
+    branches:
+      - master
   pull_request:
-    branches: [ master ]
+    branches:
+      - master
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron:  '0 0 * * 2'
 
-  # Allows you to run this workflow manually from the Actions tab
-  workflow_dispatch:
+env:
+  DOCKER_REPO: godatadriven/python-onbuild
 
-# A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  # This workflow contains a single job called "build"
   build:
-    # The type of runner that the job will run on
+    name: build
     runs-on: ubuntu-latest
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
+    strategy:
+      matrix:
+        DOCKER_TAG: [
+          "conda-python2.7",
+          "conda-python3.6",
+          "conda-python3.7",
+          "conda-python3.8",
+          "conda-python3.9,conda",
+          "pip-python2.7",
+          "pip-python3.6",
+          "pip-python3.7",
+          "pip-python3.8",
+          "pip-python3.9,pip,latest"
+        ]
+    timeout-minutes: 30
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Prepare
+        id: prep
+        run: |
+            DOCKER_TAG="${{ matrix.DOCKER_TAG }}"
+            IMAGE_TAG=${DOCKER_TAG%%,*}
+
+            DEP_TYPE=${IMAGE_TAG%-*}
+            DOCKERFILE="Dockerfile_${DEP_TYPE}"
+            echo ::set-output name=dockerfile::${DOCKERFILE}
+
+            PYTHON_TAG=${IMAGE_TAG#*-}
```

Comment:
This bash foo seems a bit funky to me. The idea behind a matrix is that you can do all kinds of combinations of tags. I would prefer something like:
```
strategy:
  matrix:
    base: ["pip-python", "conda-python"]
    version: ["2.7", "3.6", "3.7", "3.8", "3.9,pip,latest"]
```
I don't like the splitting on commas, but having the matrix above reduces the amount of splitting in bash already :)

Reply:
Yh the tags could have been done better, problem is that the tags were already setup this way so we cannot change them now.

Reply:
Mis interpreted your comment, yes this is indeed a lot nicer!

Reply:
Will push a fix later today.

Reply:
We can only have 1 latest version... Hence, this approach wont work

Reply:
Well you could have it skip steps on for example the `conda-python` and `latest` combination. But this would increase the complexity.

Reply:
`latest` is an anti-pattern anyway 💃  https://vsupalov.com/docker-latest-tag/

Reply:
Using latest is, building a latest image isn't in my opinion. As it indicates that a particular version is the image is the recommended one to use